TreeTracker/UI/Navigation/Home/HomeCoordinator.swift

```diff
@@ -116,20 +117,27 @@ private extension HomeCoordinator {
         }()
         return viewcontroller
     }
-
+    
     func profileViewController(planter: Planter) -> UIViewController {
-        let viewController = UIViewController()
-        viewController.view.backgroundColor = .white
+        let viewController = StoryboardScene.Profile.initialScene.instantiate()
+        viewController.viewModel = {
+            let treeMonitoringService = LocalTreeMonitoringService(coreDataManager: coreDataManager)
+            let selfieService = LocalSelfieService(
+                coreDataManager: coreDataManager,
+                documentManager: DocumentManager()
+            )
+            let viewModel = ProfileViewModel(planter: planter,
+                                             treeMonitoringService: treeMonitoringService, selfieService: selfieService,
+                                             uploadManager: uploadManager)
+            viewModel.coordinatorDelegate = self
+            return viewModel
+        }()
         viewController.title = {
```

Comment:
I don't think this logic should live in the coordinator. Maybe it best in the ViewModel what do you think?

Reply:
All Viewcontroller functions are written in HomeCoordinator. Eg: homeViewController, addTreeViewController, etc

Reply:
I have moved the codes for title to ProfileViewModel